### DISCUSSION TITLE
post/css-in-ts-vanilla-extract/

### DISCUSSION THREAD

**Discussion by giscus** [jgjgill/blog#24] [Comments]
# post/css-in-ts-vanilla-extract/

vanilla-extract에 대해 정리합니다.

https://jgjgill.com/post/css-in-ts-vanilla-extract/

**Comment by dlgnswk** [comment]
좋은 글 잘 읽었습니다!
저는 현재 tailwind css를 사용하고 있는데 vanilla extract로 변경을 고려하고 있습니다.
혹시 vanilla extract 사용하시면서 다른 css 라이브러리와 비교했을때 불편했던 점은 없으셨을까요?

중간에 오타가 있는것 같습니다.
벤더 프리~식~스(Vend~e~r Prefix) -> 벤더 프리픽스(Vendor Prefix)

**Reply by jgjgill** [reply]
@dlgnswk 오 좋은 말씀 감사합니다..! 🙇‍♂️

> tailwind css → vanilla extreact
> vanilla extract 다른 css 비교 불편한 점

우선 먼저 말씀드리고 싶은 부분은 저같은 경우 `vanilla extract`를 사용한 경험은 혼자 학습 목표로 사용해 본 경험이 다네요.. 🫠
그래도 한 번 질문에 답해보겠습니다..! 😇

먼저 궁금한 점은 `taillwind`에서 교체하려는 이유가 궁금하긴 하네요..! (가독성? 진입장벽? 그냥 선호도?)
(+ 마이그레이션 비용도 만만치 않을 것 같다는 생각이 들어서?!)

깊게 사용해보지는 못했지만 `tailwind`를 사용했던 경험과 비교해서 `vanilla extract` 불편한 점을 뽑아보면.. 아무래도 **이름을 짓는 것**과 **`styles.ts` 파일을 계속 만드는 것**이 아닐까 싶네요..
저는 `tailwind`에서 바로바로 스타일을 넣는 방식이 개인적인 선호도가 높긴 했네요.. ㅎㅎ

실무에서는 `styled-components`를 사용하고 있는데.. 이부분은 `zero-runtime` 측면에서 `vanilla extract`가 무조건 좋지 않을까 싶네요.. 😅

> 중간에 오타가 있는것 같습니다.
> 벤더 프리식스(Vender Prefix) -> 벤더 프리픽스(Vendor Prefix)

꼼꼼하게 살펴봐주셔서 감사합니다! 🫡  0929758742e9b2ae6b4191c2b2317eef1e563606

**Reply by dlgnswk** [reply]
답변 감사합니다!!😊

저도 `tailwind`가 더 익숙하지만, 길어지는 스타일때문에 발생하는 코드 가독성 문제와  
특유의 문법 때문에 `tailwind`가 익숙하지 않은 팀원들의 피드백 때문에 전환을 고민하게 되었어요.

대부분의 피드백은 스타일을 수정하거나 확인하기 위해서 `tailwind`의 문법을 찾아보거나 학습하는 등의 리소스를 소모하는데에서 기인한다고 생각해서
css 사용방법은 기본을 유지하면서 현대 개발방법을 잘 따라가는 라이브러리를 찾다가 `vanilla extract`를 생각하게 됐어요.

말씀 해주신 것처럼 저도 마이그레이션 비용을 고려하는 것이 중요하다고 생각되는데, 다행히도 이제 시작하고 있는 프로젝트라서  
빠르게 변경할 수 있고 전환에 있어 많은 비용이 소모되지는 않을것 같아요.

팀원들과 의견을 더 나눠봐야겠지만 작성해주신 글에서와 같이 ts와 결합이 잘되어있는 부분이나 타입 추론을 통한 스타일 적용같은 편의성이  
개발 친화적인 도구라는 생각이 들어서 결정에 많은 도움이 되었어요!

친절하게 답변 남겨주셔서 감사하고 좋은 글도 감사합니다.
앞으로도 좋은 글 기대할게요! :) 👍
